File: backend/backend/src/main/java/com/todo/api/service/TaskService.java
package com.todo.api.service;

import com.todo.api.model.Task;
import com.todo.api.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class TaskService {
    
    @Autowired
    private TaskRepository taskRepository;
    
    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }
    
    public Task saveTask(Task task) {
        return taskRepository.save(task);
    }
    
    public Optional<Task> getTaskById(Long id) {
        return taskRepository.findById(id);
    }
    
    public void deleteTask(Long id) {
        taskRepository.deleteById(id);
    }
    
    public Task updateTask(Long id, Task taskDetails) {
        Task task = taskRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Task not found with id " + id));
        
        task.setTitle(taskDetails.getTitle());
        task.setDescription(taskDetails.getDescription());
        task.setCompleted(taskDetails.isCompleted());
        
        return taskRepository.save(task);
    }
}

File: frontend/Frontend/src/components/TaskService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/tasks';

const TaskService = {
  getAllTasks: () => {
    return axios.get(API_URL);
  },
  
  createTask: (task) => {
    return axios.post(API_URL, task);
  },
  
  updateTask: (id, task) => {
    return axios.put(`${API_URL}/${id}`, task);
  },
  
  deleteTask: (id) => {
    return axios.delete(`${API_URL}/${id}`);
  }
};

export default TaskService; 